File: backend/app/Models/DetectionEvent.php
<?php

namespace App\Models;

use App\Enums\WasteType;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class DetectionEvent extends Model
{
    use HasFactory;

    protected $fillable = [
        'bin_session_id',
        'waste_type',
        'input_method',
        'detected_brand_id',
        'confidence',
        'image_path',
        'ai_output',
    ];

    protected function casts(): array
    {
        return [
            'waste_type' => WasteType::class,
            'confidence' => 'integer',
            'ai_output' => 'array',
        ];
    }

    public function binSession(): BelongsTo
    {
        return $this->belongsTo(BinSession::class);
    }

    public function detectedBrand(): BelongsTo
    {
        return $this->belongsTo(Brand::class, 'detected_brand_id');
    }

    /**
     * Compat accessor: old schema had detected_at, new uses created_at.
     */
    public function getDetectedAtAttribute(): ?\Illuminate\Support\Carbon
    {
        return $this->created_at;
    }

    /**
     * Compat accessor: old views reference $detection->bin directly.
     * New schema goes through binSession.
     */
    public function getBinAttribute(): ?Bin
    {
        return $this->binSession?->bin;
    }

    /**
     * Weekly detection chart data.
     */
    public static function weeklyChart(?Bin $bin = null): array
    {
        $labels = [];
        $data = [];

        for ($i = 6; $i >= 0; $i--) {
            $date = now()->subDays($i);
            $labels[] = $date->format('D');
            $query = static::whereDate('created_at', $date);
            if ($bin) {
                $query->whereHas('binSession', fn ($q) => $q->where('bin_id', $bin->id));
            }
            $data[] = $query->count();
        }

        return ['labels' => $labels, 'data' => $data];
    }
}

File: mobile/Mobius/Sources/Models/DetectionEvent.swift
import Foundation

struct DetectionEvent: Codable, Identifiable, Sendable {
    let id: Int
    var binId: Int
    var userId: Int?
    var wasteType: WasteType?
    var confidence: Int?
    var weightG: Int?
    var detectedBrandId: Int?
    var imagePath: String?
    var pointsEarned: Int?
    var detectedBrand: DetectedBrand?
    var createdAt: Date?

    enum CodingKeys: String, CodingKey {
        case id
        case binId = "bin_id"
        case userId = "user_id"
        case wasteType = "waste_type"
        case confidence
        case weightG = "weight_g"
        case detectedBrandId = "detected_brand_id"
        case imagePath = "image_path"
        case pointsEarned = "points_earned"
        case detectedBrand = "detected_brand"
        case createdAt = "created_at"
    }
}

struct DetectedBrand: Codable, Identifiable, Sendable {
    let id: Int
    let name: String
    let slug: String
    var logoPath: String?

    enum CodingKeys: String, CodingKey {
        case id, name, slug
        case logoPath = "logo_path"
    }
}
